foundation-ui/routing/RoutingService.js

```diff
@@ -0,0 +1,148 @@
+import {Route, Redirect} from 'react-router';
+import {EventEmitter} from 'events';
+
+import {ROUTING_CHANGE} from './EventTypes';
+
+function throwError(error) {
+  if (global.__DEV__) {
+    throw error;
+  }
+}
+
+class RoutingService extends EventEmitter {
+
+  constructor() {
+    super();
+
+    const privateContext = {
+      instance: this,
+      definition: [],
+      deferredTasks: [],
+      defer(args) {
+        this.deferredTasks.push(args);
+      },
+      processDeferred() {
+        const tasks = this.deferredTasks.slice(0);
+        this.deferredTasks = [];
+
+        // If Task is unable to resolve at this point of time it will re-defer itself
+        tasks.forEach((args) => {
+          this.instance.registerTab.apply(this.instance, args);
+        });
+      }
+    };
+
+    this.getDefinition = this.getDefinition.bind(privateContext);
```

Comment:
I would suggest to add a `@this` JSDoc comment on the `getDefinition`, `registerPage`, `registerTab` and `registerRedirect` comment blocks.